R/html.R

```diff
@@ -440,6 +444,40 @@ split_chapters = function(output, build = build_chapter, number_sections, split_
   nms[j]
 }
 
+build_404_page <- function(html_head, html_toc, html_foot, build, ...) {
+  path_404 = "404.html"
+  # if a 404 page already exist, we do nothing specific and assume
+  # user has already a workflow in place
+  if (file.exists(path_404)) return()
+  # We create 404 page if it does not exist
+  if (file.exists(found <- "_404.md") || file.exists(found <- "_404.Rmd")) {
+    rmd_cur = found
+    xfun::Rscript_call(function() {
+      rmarkdown::render(found,
+                        rmarkdown::html_fragment(
+                          pandoc_args = c("--metadata","title=404")
+                        ),
+                        output_file = path_404,
+                        quiet = TRUE
+      )
+    })
+    html_404 = Filter(nzchar, xfun::read_utf8(path_404)) # remove empty line
```

Comment:
What's the goal of removing empty lines?

Reply:
Some empty lines are introduced at the top of the `404.html` created with `html_fragment()`. I believe this is because of the template for `html_fragment()` which has empty lines and if conditional. 

So I wanted to remove them are they are not needed.

Reply:
In this case, these empty lines don't really matter, right?

Sometimes empty lines do matter, e.g., inside `<pre></pre>`. I tend not to apply `Filter(nzchar)` here. What do you think?

Reply:
I see. Empty lines seems to not be an issue when in source code - The browser will remove them when they are between HTML tags. I'll remove this then.

Reply:
Done in a8d98d9